### PR TITLE
[v4] 19/07/24 Merge master into v4

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### What

Merging master into version 4.

The last commit has the changes to fix the build/errors after conflicts were resolved.

### Percy

Percy is creating a lot of noise so I'm not approving the changes till reviews have had a chance to 👀 it. From what I saw it's nothing to worry about.

#### Underline changes

They are from #1686, here's the [percy report](https://percy.io/Shopify/polaris-react/builds/2037363?utm_campaign=Shopify&utm_content=polaris-react&utm_source=github_status_public)

#### Page header changes

They are from #1653, here's the [percy report](https://percy.io/Shopify/polaris-react/builds/2024750?utm_campaign=Shopify&utm_content=polaris-react&utm_source=github_status_public)

#### ResourceList / Button

New/update examples have cause shifted

#### Card

Percy bug? There were some new examples. Not sure why these all came up?

### How to 🎩

* I usually like to compare [v4...this_branch](https://github.com/Shopify/polaris-react/compare/master...19-07-24-master-to-v4?expand=1) -> master...this_branch but everyone has their own methods.
* tests 💯 ?
* percy ✅ ?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,14 +97,7 @@
     "@shopify/js-uploader": "github:shopify/js-uploader",
     "@shopify/react-testing": "^1.5.3",
     "@shopify/sewing-kit": "0.88.0",
-    "@storybook/addon-a11y": "5.1.0-rc.0",
```
Comment (AndrewMusgrave) [inline package.json]: Looks like these might have accidentally be moved to dependencies, I'm going to follow up to Kaelig.

File: sewing-kit.config.ts [hunk 2]
```diff
@@ -11,11 +15,11 @@ export default function sewingKitConfig(
     name: 'polaris',
     library: true,
     plugins: [
-      plugins.jest((config) => {
+      plugins.jest((config: InitialOptions) => {
         config.roots = [join(__dirname, 'src'), join(__dirname, 'tests')];
 
         config.setupFiles.push(join(tests, 'setup.ts'));
-        config.setupTestFrameworkScriptFile = join(tests, 'each-test.ts');
```
Comment (AndrewMusgrave) [inline sewing-kit.config.ts]: This is now deprecated and replaced with `setupFilesAfterEnv` which accepts multiple files

File: src/components/AppProvider/tests/AppProvider.test.tsx [hunk 3]
```diff
@@ -7,7 +7,10 @@ describe('<AppProvider />', () => {
   it('updates polaris context when props change', () => {
     const Child: React.SFC<{}> = (_props) => (
       <AppProviderContext.Consumer>
-        {({link}) => (link.getLinkComponent() ? <div id="child" /> : null)}
+        {({link}) => {
+          // eslint-disable-next-line shopify/jest/no-if
```
Comment (AndrewMusgrave) [inline src/components/AppProvider/tests/AppProvider.test.tsx]: We're getting this false positive a lot, but we're already aware of the issue

**Comment by AndrewMusgrave** [comment]
If anyone has a chance today could they give this some love? It's a blocker ✌️🙇‍♂️

**Comment by AndrewMusgrave** [comment]
Looking at master reports card have individual sections now

**Comment by AndrewMusgrave** [comment]
@BPScott Unfortunately merging master will be a little difficult. Some deprecated/removed code is now being used which will require some conversions. I'll work on merging master separately immediately after this is merged

File: examples/create-react-app/yarn.lock [hunk 4]
```diff
@@ -9728,14 +9733,22 @@ typedarray@^0.0.6:
...
   integrity sha1-hnrHTjhkGHsdPUfZlqeOxciDB3c=
 
-uglify-js@3.4.x, uglify-js@^3.1.4:
+uglify-js@3.4.x:
   version "3.4.9"
   resolved "https://registry.yarnpkg.com/uglify-js/-/uglify-js-3.4.9.tgz#af02f180c1207d76432e473ed24a28f4a782bae3"
   integrity sha512-8CJsbKOtEbnJsTyv6LE6m6ZKniqMiFWmm9sRbopbkGs3gMPPfd3Fh8iIA4Ykv5MgaTbqHr4BaoGLJLZNhsrW1Q==
   dependencies:
     commander "~2.17.1"
     source-map "~0.6.1"
```
Comment (dleroux) [inline examples/create-react-app/yarn.lock]: Do we know why this time uglify produced 2 entries?

**Comment by dleroux** [comment]
This looks good. A bit hard to review because of all the changes in master. Might be worth highlighting the changes you make to accommodate the merge if any. As for Percy that is bizarre. We'll definitely need a thorough visual top hat when this is ready.

**Comment by AndrewMusgrave** [comment]
> This looks good. A bit hard to review because of all the changes in master. Might be worth highlighting the changes you make to accommodate the merge if any. As for Percy that is bizarre. We'll definitely need a thorough visual top hat when this is ready.

I went over my changes and the percy report in the description 😄

File: examples/create-react-app/yarn.lock (hunk 4 above)
Comment (AndrewMusgrave) [reply examples/create-react-app/yarn.lock]: It's removing an entry, looks it like was added [here](https://github.com/Shopify/polaris-react/commit/8f7c64bde4f501de9b2c20f82b4182c35172659a#diff-8ee2343978836a779dc9f8d6b794c3b2R17598). To quiet `audits`, but a version was probably bumped to be in semver range of `uglify-js@3.4.x:`

**Pull request merged**